lib/commands/exam.js

```diff
@@ -254,5 +372,97 @@ module.exports = TestCommand.extend({
       case 'json':
         return fs.readJsonSync(file);
     }
+  },
+
+  /**
+   * Adds additional event handlers to config to enable load balancing.
+   *
+   * @param {object} config
+   * @return {object} events
+   */
+  _addLoadBalancingBrowserSocketEvents(config) {
+    const events = config.custom_browser_socket_events || {};
+
+    events['set-modules-queue'] = function(modules) {
+      // When `load-balance` option is valid we want to have one static list of modules on server side to send a module path to browsers.
+      const proto = Object.getPrototypeOf(this);
```

Comment:
Using the prototype for storing/sharing state between these two events is not appropriate. Prototypes are general reserved for inherited properties and methods.

If we have explicit state, lets create a new object for said state, with clear methods for interaction. That way it is explicit, obvious, maintainable and testable.

I'll wait until this area is refactored as described above before continuing the review.

Reply:
@stefanpenner the event handlers here gets registered on testem's browser runner's socket connection: https://github.com/testem/testem/commit/209920ff3e523dc74b1f22e98ac3415b59a2eae7
So the state needs to be saved within testem's scope.

Within testem, there is no official place to store test state. With `config` being a shared object that all runner instance has access to. Would using `config` for this purpose be reasonable?

Reply:
Resolved this by adding a execution-state-manager